lib/std/math/nan.zig

```diff
@@ -2,6 +2,18 @@ const math = @import("../math.zig");
 
 /// Returns the nan representation for type T.
 pub inline fn nan(comptime T: type) T {
+    return switch (@typeInfo(T).Float.bits) {
+        16 => math.qnan_f16,
+        32 => math.qnan_f32,
+        64 => math.qnan_f64,
+        80 => math.qnan_f80,
+        128 => math.qnan_f128,
+        else => @compileError("unreachable"),
+    };
+}
```

Comment:
This change should be incorporated via the separate PR https://github.com/ziglang/zig/pull/16826.